geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/BatchWriter.scala

```diff
@@ -98,10 +94,9 @@ object BatchWriter {
     }
 
     // this is lazy to allow the query plan to be instantiated without pulling back all the batches first
-    // we also allow for early termination by checking the `closed` flag
     private lazy val inputs: Array[(SimpleFeatureVector, (Int, Int) => Unit)] = {
       val builder = Array.newBuilder[(SimpleFeatureVector, (Int, Int) => Unit)]
-      while (!closed.get && batches.hasNext) {
```

Comment:
Ok, why can we get away with this at this point?

Reply:
`batches.next` will throw an exception now once it's been terminated, so we don't need to check the closed flag (and closed won't get invoked by the thread management anyway)

Reply:
What happens if someone gets in there and calls close separately?  

Does that even matter?

Reply:
hmm, yeah that would not work well since it would block on the lazy val again... maybe i should add the closed check back in

Reply:
actually nvm, the close method closes the batches so it wouldn't block. let me go through and double check that in the different places we had synchronization.